pkg/controllers/sync/dispatch/managed.go

```diff
@@ -84,8 +72,8 @@ type ManagedDispatcher interface {
 	Update(ctx context.Context, clusterName string, clusterObj *unstructured.Unstructured)
 	VersionMap() map[string]string
 	CollectedStatus() status.CollectedPropagationStatus
-	RecordClusterError(propStatus fedtypesv1a1.PropagationStatus, clusterName string, err error)
-	RecordStatus(clusterName string, propStatus fedtypesv1a1.PropagationStatus)
+	RecordClusterError(propStatus fedcorev1a1.PropagationStatusType, clusterName string, err error)
```

Comment:
Can we merge `RecordClusterError` and `RecordStatus` since all `RecordClusterError` is doing now is to emit an event.